javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt

```diff
@@ -0,0 +1,60 @@
+package arcs.core.crdt
+
+import arcs.core.testutil.ChooseFromList
+import arcs.core.testutil.GetInt
+import arcs.core.testutil.MapOf
+import arcs.core.testutil.SetOf
+import arcs.core.testutil.Value
+import arcs.core.testutil.runFuzzTest
+import com.google.common.truth.Truth
+import org.junit.Test
+
+class CrdtEntityFuzzTest() {
+  /**
+   * Test that a randomly generated [CrdtEntity] merged with itself produces an empty [CrdtChange].
+   */
+  @Test
+  fun crdtEntity_merge_invariant() = runFuzzTest {
+    val versionMap = VersionMapGenerator(
+      ChooseFromList(it, listOf("me", "foo", "bar", "fooBar")),
+      GetInt(it, 1, 20)
+    )
+
+    val singletons = MapOf(
+      ChooseFromList(it, listOf("koala", "kangaroo", "penguin", "emu")),
+      ReferencableGenerator(ChooseFromList(it, listOf("animal", "mammal", "reptile", "shelled"))),
+      GetInt(it, 0, 3)
+    )
+
+    val collections = MapOf(
+      ChooseFromList(it, listOf("koalas", "kangaroos", "penguins", "emus")),
+      SetOf(
+        ReferencableGenerator(
+          ChooseFromList(
+            it,
+            listOf("animals, mammals, reptiles", "shelleds", "pets", "birds", "fish")
```

Comment:
should animals, mammals, reptiles be a single option, or 3 separate options? A comment might be good if they're meant to be a single option.

Reply:
Oops, it should be 3 separate options. I'm not sure how that happened....